src/lib/krb5/krb/preauth_sam2.c

```diff
@@ -150,7 +150,6 @@ sam2_process(krb5_context context, krb5_clpreauth_moddata moddata,
         /* Go ahead and get it now, preserving the ordering of  */
         /* prompts for the user.                                */
 
-        salt = ctx->default_salt ? NULL : &ctx->salt;
```

Comment:
This one is weird, and probably requires a closer look.

The gak_fct() call here should pass salt instead of &ctx->salt, so that krb5_get_as_key_password() computes and uses the default salt instead of using an empty salt, when the KDC didn't send an explicit salt.  That's a bug in commit 787cc1e2411 or thereabouts, and the unused value of salt is correctly flagged by Coverity.

However, the result of this gak_fct() call is ignored, because at line 224 we explicitly call krb5_c_string_to_key() on the password with the salt we correctly determined at lines 200-211.  That apparently redundant operation dates back much longer.

I would like to fix the gak_fct() call and then eliminate the krb5_c_string_to_key() call at line 224.  I will need to do some testing.

We unfortunately still need lines 200-211 as long as we support either of the non-SEND_ENCRYPTED_SAD cases.

Reply:
I believe you've superseded this with https://github.com/krb5/krb5/pull/924 so dropped from this series.